### PR TITLE
instance_eval if option in attributes

### PR DESCRIPTION
**Pull request by hallgren**
A new PR for #116 fixing the commit messages.

### CODE REVIEW

File: yaks/lib/yaks/mapper/attribute.rb [hunk 1]
```diff
@@ -2,22 +2,21 @@ module Yaks
...
       extend Forwardable, Util
-      include Util
-      include Attribs.new(:name, :block, options: {}.freeze)
+      include Attribs.new(:name, :block, if: true), Util
 
-      def self.create(*args, &block)
-        args, options = extract_options(args)
-        new(name: args.first, options: options, block: block)
+      def self.create(name, options = {}, &block)
+        new(options.merge({name: name, block: block}))
```
Comment (groyoh) [inline yaks/lib/yaks/mapper/attribute.rb]: The `{` and `}` are not needed :wink:

File: yaks/lib/yaks/mapper/attribute.rb [hunk 2]
```diff
@@ -2,22 +2,21 @@ module Yaks
...
       end
 
       def add_to_resource(resource, mapper, _context)
-        return unless mapper.expand_value(options.fetch(:if, true))
-
+        return resource unless Resolve(self.if, mapper)
+
```
Comment (groyoh) [inline yaks/lib/yaks/mapper/attribute.rb]: Extra whitespaces here. :wink:

File: yaks/lib/yaks/mapper/attribute.rb [hunk 3]
```diff
@@ -2,22 +2,21 @@ module Yaks
...
 
-      def self.create(*args, &block)
-        args, options = extract_options(args)
-        new(name: args.first, options: options, block: block)
+      def self.create(name, options = {}, &block)
+        new(options.merge({name: name, block: block}))
       end
 
       def add_to_resource(resource, mapper, _context)
-        return unless mapper.expand_value(options.fetch(:if, true))
```
Comment (groyoh) [inline yaks/lib/yaks/mapper/attribute.rb]: Is there any reason why you changed `mapper.expand_value` to `Resolve`?

**Comment by groyoh** [comment]
It would have been better to squash the commits instead of opening a new PR. Could you amend the commit and add a better description of the change?

Ex:

```
Add specs for Yaks::Mapper::Attribute

The specs now test that the "if" option of Yaks::Mapper::Attribute accepts procs and lambdas.
```

**Comment by hallgren** [comment]
To do that I had to rewrite the history of an already pushed branch.
Thats why I created the new PR.

File: yaks/lib/yaks/mapper/attribute.rb (hunk 3 above)
Comment (hallgren) [inline yaks/lib/yaks/mapper/attribute.rb]: If i remember correctly mapper.expand_value could not access the `object` in the Proc

Comment (groyoh) [inline yaks/lib/yaks/mapper/attribute.rb]: That's weird given that `mapper.expand_value(value)` simply calls `Resolve(value, self)` (ref. https://github.com/plexus/yaks/blob/master/yaks/lib/yaks/mapper.rb#L68)...

**Comment by groyoh** [comment]
It's usually ok to rewrite the history (rebase) of your own branch (even if it's pushed) as long as nobody else is working on it and it isn't merged. In the case of PRs, it's often recommended to squash before merging to keep history clean of WIP commits. And that's usually done by rebasing and force pushing ;) This also avoid creating too many PRs.

**Comment by hallgren** [comment]
@groyoh I made the adjustment you wanted. Is it ok?

**Comment by groyoh** [comment]
The PR looks great :+1: Just need to squash it as described in https://github.com/plexus/yaks/pull/117#issuecomment-199753388

**Comment by hallgren** [comment]
@groyoh squashed :)

**Comment by groyoh** [comment]
Thanks for rebasing. As requested many times by @plexus and I, could you please provide a better commit message? `instance_eval if option` isn't actually what the commit does and is not a proper commit message. Check the example I gave you at https://github.com/plexus/yaks/pull/117#issuecomment-199753388

**Comment by hallgren** [comment]
@groyoh I have now changed the commit message to what I think it does, thanks for your comment.

**Comment by groyoh** [comment]
Thanks for the PR :+1: I'm merging it.

**Comment by hallgren** [comment]
@groyoh Are there any plan in publish i public release to rubygems?

**Comment by groyoh** [comment]
According to @plexus [comment](https://github.com/plexus/yaks/pull/116#issuecomment-196874998), he will probably put out a new release. :wink:
@plexus could you confirm that?

**Comment by plexus** [comment]
Released as v0.12.0, thanks for all the help!

**Comment by hallgren** [comment]
nice! thanks

**Pull request merged**